src/jit/gentree.h

```diff
@@ -1425,6 +1426,20 @@ struct GenTree
                || OperIsShiftOrRotate(op);
     }
 
+#ifdef _TARGET_XARCH_
+    static bool OperIsRMWMemOp(genTreeOps gtOper)
+    {
+        // Return if binary op is not one of the supported operations for RMW of memory.
+        return (gtOper == GT_ADD || gtOper == GT_SUB || gtOper == GT_AND || gtOper == GT_OR || gtOper == GT_XOR ||
```

Comment:
Oops, I copied the comment from the place I am now calling it from, and it was the negation. Will Fix.